src/file_proxy_s3.cc

```diff
@@ -276,6 +276,32 @@ s3_delete_object(const char* file_path)
 	return del_buffer.Flush();
 }
 
+bool
+ListAllObjects(const Aws::S3::S3Client& client, Aws::S3::Model::ListObjectsV2Request& req, Aws::Vector<Aws::S3::Model::Object>& target) {
+
+	Aws::S3::Model::ListObjectsV2Outcome outcome;
+	do {
+		outcome = client.ListObjectsV2(req);
+		if (!outcome.IsSuccess()) {
+			err("%s", outcome.GetError().GetMessage().c_str());
+			return false;
+		}
+
+		Aws::S3::Model::ListObjectsV2Result res = outcome.GetResult();
+		if (res.GetIsTruncated()) {
+			const Aws::String& ct = res.GetNextContinuationToken();
+			req.SetContinuationToken(ct);
+		}
+
+		for (const Aws::S3::Model::Object& object : res.GetContents()) {
+			target.push_back(object);
+		}
+	}
+	while (outcome.GetResult().GetIsTruncated());
```

Comment:
might not be possible, but maybe save the result of `outcome.GetResult().GetIsTruncated()` since you use it in the loop.  I am not sure if those function calls are idempotent or not

Reply:
The only functions called on `res` are getters that are idempotent.